tests/test_uninstall.py

```diff
@@ -0,0 +1,432 @@
+    shortcut_env = base_env / "envs" / "shortcutenv"
+    run_conda("create", "-y", "-p", str(shortcut_env), *shortcuts)
+    assert _shortcuts_found(shortcut_env) == shortcuts
+    run_uninstaller(base_env)
+    assert _shortcuts_found(shortcut_env) == []
+
+
+@pytest.mark.parametrize(
+    "shared_pkgs",
+    (True, False),
+    ids=("shared pkgs", "remove pkgs"),
+)
+def test_uninstallation_remove_caches(
+    mock_system_paths: dict[str, Path],
+    tmp_env: TmpEnvFixture,
+    shared_pkgs: bool,
+):
+    # Set up notices
+    if ON_WIN:
+        try:
+            import ctypes
+
+            if not hasattr(ctypes, "windll"):
+                pytest.skip("Test requires windll.ctypes for mocked locations to work.")
+        except ImportError:
+            pytest.skip("Test requires ctypes for mocked locations to work.")
+        notices_dir = Path(
+            mock_system_paths["cachehome"], "conda", "conda", "Cache", "notices"
+        )
+    else:
+        notices_dir = Path(mock_system_paths["cachehome"], "conda", "notices")
+    notices_dir.mkdir(parents=True, exist_ok=True)
+    (notices_dir / "notices.cache").touch()
+
+    yaml = YAML()
+    pkgs_dir = mock_system_paths["home"] / "pkgs"
+    condarc = {
+        "channels": [CONDA_CHANNEL],
+        "pkgs_dirs": [str(pkgs_dir)],
+    }
+    with open(mock_system_paths["home"] / ".condarc", "w") as crc:
+        yaml.dump(condarc, crc)
+
+    other_env = tmp_env("python") if shared_pkgs else nullcontext()
+    with (
+        tmp_env("constructor") as base_env,
+        other_env as _,
+    ):
+        assert pkgs_dir.exists()
+        assert list(pkgs_dir.glob("constructor*")) != []
+        assert list(pkgs_dir.glob("python*")) != []
+        run_uninstaller(base_env, remove_caches=True)
+        assert pkgs_dir.exists() == shared_pkgs
+        if shared_pkgs:
+            assert list(pkgs_dir.glob("constructor*")) == []
+            assert list(pkgs_dir.glob("python*")) != []
+        assert not notices_dir.exists()
+
+
+def test_uninstallation_remove_user_data(
+    mock_system_paths: dict[str, Path],
+    tmp_env: TmpEnvFixture,
+):
+    with tmp_env() as base_env:
+        dot_conda_dir = mock_system_paths["home"] / ".conda"
+        assert dot_conda_dir.exists()
+        run_uninstaller(base_env, remove_user_data=True)
+        assert not dot_conda_dir.exists()
+
+
+@pytest.mark.parametrize("remove", ("user", "system", "all"))
+@pytest.mark.skipif(not ON_CI, reason="CI only - Writes to system files")
+def test_uninstallation_remove_config_files(
+    mock_system_paths: dict[str, Path],
+    tmp_env: TmpEnvFixture,
+    remove: str,
+):
+    yaml = YAML()
+    remove_system = remove != "user"
+    remove_user = remove != "system"
+    condarc = {
+        "channels": [CONDA_CHANNEL],
+    }
+    needs_sudo = False
+    if ON_WIN:
+        system_condarc = Path("C:/ProgramData/conda/.condarc")
+    else:
+        system_condarc = Path("/etc/conda/.condarc")
```

Comment:
Should we rely on env vars like `PROGRAMDATA` and `XDG_DATA_*` for this test? Or do you think it's not necessary?

Reply:
I 100% agree that we should rely on `PROGRAMDATA`, but unfortunately, `conda` doesn't use those variables: https://github.com/conda/conda/blob/main/conda/base/constants.py#L29-L42